main.tf

```diff
@@ -1 +1,200 @@
+    resources = ["*"]
+  }
+
+  statement {
+    sid       = "ListLogFiles"
+    resources = ["arn:aws:s3:::aws-controltower-logs-${var.aws_log_archive_account_id}-${var.control_tower_region}/${var.aws_organization_id}/*AWSLogs/*"]
+    actions   = ["s3:ListBucket"]
+
+    condition {
+      test     = "StringLike"
+      variable = "s3:prefix"
+      values   = ["*AWSLogs/"]
+    }
+  }
+
+  statement {
+    sid       = "ConsumeNotifications"
+    # resources = ["${aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn}"]
+    resources = flatten([var.sqs_queues, [aws_sqs_queue.lacework_cloudtrail_sqs_queue.arn]])
+    actions = [
+      "sqs:GetQueueAttributes",
+      "sqs:GetQueueUrl",
+      "sqs:DeleteMessage",
+      "sqs:ReceiveMessage"
+    ]
+  }
+
+  statement {
+    sid       = "Debug"
+    resources = ["*"]
+    actions = [
+      "cloudtrail:DescribeTrails",
+      "cloudtrail:GetTrail",
+      "cloudtrail:GetTrailStatus",
+      "cloudtrail:ListPublicKeys",
+      "s3:GetBucketAcl",
+      "s3:GetBucketPolicy",
+      "s3:ListAllMyBuckets",
+      "s3:GetBucketLocation",
+      "s3:GetBucketLogging",
+      "sns:GetSubscriptionAttributes",
+      "sns:GetTopicAttributes",
+      "sns:ListSubscriptions",
+      "sns:ListSubscriptionsByTopic",
+      "sns:ListTopics"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "cross_account_policy" {
+  provider = aws.log_archive
+  name        = local.cross_account_policy_name
+  description = "A cross account policy to allow Lacework to pull config and cloudtrail"
+  policy      = data.aws_iam_policy_document.cross_account_policy.json
+}
+
+module "lacework_ct_iam_role" {
+  providers = {
+    aws = aws.log_archive
+  }
+  source                  = "lacework/iam-role/aws"
+  version                 = "~> 0.1"
+  create                  = var.use_existing_iam_role ? false : true
+  iam_role_name           = local.iam_role_name
+  lacework_aws_account_id = var.lacework_aws_account_id
+  external_id_length      = var.external_id_length
+  tags                    = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "lacework_cross_account_iam_role_policy" {
+  provider = aws.log_archive
+  role       = local.iam_role_name
+  policy_arn = aws_iam_policy.cross_account_policy.arn
+  depends_on = [module.lacework_ct_iam_role]
+}
+
+# wait for X seconds for things to settle down in the AWS side
+# before trying to create the Lacework external integration
+resource "time_sleep" "wait_time" {
+  create_duration = var.wait_time
+  depends_on = [
+    aws_iam_role_policy_attachment.lacework_cross_account_iam_role_policy,
+    aws_sns_topic_subscription.lacework_sns_topic_sub
+  ]
+}
+
+resource "lacework_integration_aws_ct" "default" {
+  // do not create a CT integration if the user provides multiple
+  // SQS queues to configure, it means that they want to fan-out
+  // with a lambda function that acks like a gateway
+  count     = length(var.sqs_queues) > 0 ? 0 : 1
```

Comment:
You do not need this code since we used it a year and a half ago when there was no
concept of Org Account Mappings
```suggestion
```